docs/deviceUsecase/deviceBgpUsecase.md

```diff
@@ -0,0 +1,260 @@
+v4_routes.addresses.add(address="20.20.0.0")
+```
+- Single ethernet(“eth1”) configurate on top of port “p1”
+- Single IPv4(“ip1”) present on top of ethernet(“eth1”)
+- Single BGP map with IPv4(“ip1”)
+- Two v4_route configure on that bgp_peer
+## IxNetwork Mapping
+<img src="scr_bgp_1.png" alt="drawing" width="500"/>
+
+- Create topology per port
+- Ether, IP and BGP can map one to one
+- Create network group (NG) according to v4_routes
+- Use NG multiplier to accommodate number of address
+
+# Scenario-2: Single Interface and Multiple BGP Peer
+```python
+device = config.devices.device(name="d1")[-1]
+eth1 = device.ethernets.ethernet(name='eth1', port_name="p1")[-1]
+eth1.ipv4_addresses.ipv4(name="ip1")
+
+bgp = device.bgp
+bgp.router_id = "1.1.1.1"
+bgp_int1 = bgp.ipv4_interfaces.add(ipv4_name="ip1")
+bgp_peer1 = bgp_int1.peers.add(name="bgp1")
+v4_routes1 = bgp_peer1.v4_routes.add(name="route1")
+v4_routes1.addresses.add(address="10.10.0.0")
+bgp_int2 = bgp.ipv4_interfaces.add(ipv4_name="ip1")
+bgp_peer2 = bgp_int2.peers.add(name="bgp2")
+v4_routes2 = bgp_peer2.v4_routes.add(name="route2")
+v4_routes2.addresses.add(address="20.20.0.0")
+```
+- Single Ethernet and IP
+- Two BGP peers map to single IP
+## IxNetwork Mapping
+<img src="scr_bgp_2.png" alt="drawing" width="500"/>
+
+- Device multiplier set to 1
+- IP stack Multiplier set to 1
+- BGP stack Multiplier should set according to the number of BGP peers ("2")
+- Compact all values related BGP. And configure those in respective rows.
+
+# Scenario-3: Single Ethernet and Multiple IP and BGP Peer 
+```python
+device = config.devices.device(name="d1")[-1]
+eth1 = device.ethernets.ethernet(name='eth1', port_name="p1")[-1]
+eth1.ipv4_addresses.ipv4(name="ip1")
+eth1.ipv4_addresses.ipv4(name="ip2")
+
+bgp = device.bgp
+bgp.router_id = "1.1.1.1"
+bgp_int1 = bgp.ipv4_interfaces.add(ipv4_name="ip1")
+bgp_peer1 = bgp_int1.peers.add(name="bgp1")
+v4_routes = bgp_peer1.v4_routes.add(name="route1")
+v4_routes.addresses.add(address="10.10.0.0")
+bgp_int2 = bgp.ipv4_interfaces.add(ipv4_name="ip2")
+bgp_peer2 = bgp_int2.peers.add(name="bgp2")
+v4_routes = bgp_peer2.v4_routes.add(name="route2")
+v4_routes.addresses.add(address="20.20.0.0")
+```
+- Single Ethernet on top of single port
+- Two IP configure on top of single ethernet
+- Two BGP peers map with Two IP
+## IxNetwork Mapping
+<img src="scr_bgp_3.png" alt="drawing" width="500"/>
+
+- Device multiplier set to 1
+- IP stack Multiplier should set according to the number of IP address ("2")
+- BGP stack Multiplier set to 1
+- Compact all values related to IP and BGP. And configure those in respective rows.
+
+# Scenario-4: Multiple devices configured top of same Port
+```python
+device1 = config.devices.device(name="d1")[-1]
+device2 = config.devices.device(name="d2")[-1]
+
+eth1 = device1.ethernets.ethernet(name='eth1', port_name="p1")[-1]
+eth2 = device2.ethernets.ethernet(name='eth2', port_name="p1")[-1]
+eth1.ipv4_addresses.ipv4(name="ip1")
```

Comment:
Modify test script

Reply:
@apratimmukherjee is this use case without vlan valid at all ? if so, we'll need to mention it separately (to ensure we have TCs for it too).

Reply:
Yes, possible but unlikely without a switch between IxN and DUT. Since DUT will not allow multiple subnets on same physical port. It should work in b2b , though that does not add much value for real-life use-cases.